anesthetic/plot.py

```diff
@@ -948,6 +1000,8 @@ def hist_plot_1d(ax, data, *args, **kwargs):
     kwargs = normalize_kwargs(kwargs)
     weights = kwargs.pop('weights', None)
     bins = kwargs.pop('bins', 10)
+    if ax.get_xaxis().get_scale() == 'log' and isinstance(bins, int):
+        bins = np.logspace(np.log10(data.min()), np.log10(data.max()), bins+1)
```

Comment:
Note that these two lines are only needed when directly calling `hist_plot_1d`. When it gets called via `plot_1d` or `plot_2d`, then this is handled in [anesthetic/plotting/_matplotlib/hist.py](https://github.com/handley-lab/anesthetic/pull/328/files#diff-10c686dc66f7a34f6647a52acd441230844a8e49d2e24a6adef1f57722a908de). 

This is similar to the double range handling from https://github.com/handley-lab/anesthetic/pull/314#discussion_r1272296536.